hw2.sql

```diff
@@ -0,0 +1,161 @@
+--Formatted with https://www.dpriver.com/pp/sqlformat.htm
+
+USE baseball;
+
+--Create hits table with only hits and no outs
+CREATE TABLE only_hits AS
+  SELECT *
+  FROM   hits
+  WHERE  type = 'H';
+
+--Append local date to hits table so i dont have to join with game table every time
+CREATE TABLE hits_with_date AS
+  SELECT h.*,
+         g.local_date
+  FROM   only_hits h
+         JOIN game g
+           ON h.game_id = g.game_id;
+
+--append local date to atbats table
+CREATE TABLE atbats_with_date AS
+  SELECT ab.*,
+         g.local_date
+  FROM   atbat_r ab
+         JOIN game g
+           ON ab.game_id = g.game_id;
+
+--create hits by year table
+CREATE TABLE hits_by_year AS
+  SELECT Count(hits_id)   AS hits,
+         batter,
+         Year(local_date) AS year
+  FROM   hits_with_date
+  GROUP  BY batter,
+            year;
+
+--create atbats by year table
+CREATE TABLE atbats_by_year AS
+  SELECT Count(atbat_r_id) AS atbats,
+         batter,
+         Year(local_date)  AS year
+  FROM   atbats_with_date
+  GROUP  BY batter,
+            year;
+
+--create batting averge by year table
+CREATE TABLE batting_avg_by_year AS
```

Comment:
Adding ab.[field] might make things a bit easier to read (as it's explicit about which table the data is coming from).